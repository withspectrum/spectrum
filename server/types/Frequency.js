const Frequency = /* GraphQL */ `
	type FrequencySubscribersConnection {
		pageInfo: PageInfo!
		edges: [FrequencySubscriberEdge!]
	}

	type FrequencySubscriberEdge {
		cursor: String!
		node: User!
	}

	type FrequencyStoriesConnection {
		pageInfo: PageInfo!
		edges: [FrequencyStoryEdge!]
	}

	type FrequencyStoryEdge {
		cursor: String!
		node: Story!
	}

	type FrequencyMetaData {
		stories: Int
		subscribers: Int
	}

	type Frequency {
		id: ID!
		createdAt: Date!
		modifiedAt: Date
		name: String!
		description: String!
		slug: String!
		community: Community!
		storyConnection(first: Int = 10, after: String): FrequencyStoriesConnection!
		subscriberConnection(first: Int = 10, after: String): FrequencySubscribersConnection!
		subscriberCount: Int!
		metaData: FrequencyMetaData
	}

	extend type Query {
		frequency(id: ID, slug: String, community: String): Frequency
		topFrequencies(amount: Int = 30): [Frequency!]
	}
`;

module.exports = Frequency;
