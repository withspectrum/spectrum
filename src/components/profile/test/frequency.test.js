// @flow
import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Frequency from '../frequency';

const data = {
  title: 'Test frequency',
  subtitle: 'Whats up',
  description: 'Testing the frequency profile',
  id: 'asdf-123',
};

it('should render with no size', () => {
  const wrapper = shallow(<Frequency data={data} />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});

it('should render with a mini size', () => {
  const wrapper = shallow(<Frequency size="mini" data={data} />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});

it('should render with a small size', () => {
  const wrapper = shallow(<Frequency size="small" data={data} />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});

it('should render with a medium size', () => {
  const wrapper = shallow(<Frequency size="medium" data={data} />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});

it('should render with a large size', () => {
  const wrapper = shallow(<Frequency size="large" data={data} />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});

it('should render with a full size', () => {
  const wrapper = shallow(<Frequency size="full" data={data} />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});
