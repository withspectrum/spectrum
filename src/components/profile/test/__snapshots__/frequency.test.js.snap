exports[`test should render with a full size 1`] = `
<pure(CardPure)
  data={
    Object {
      "description": "Testing the frequency profile",
      "id": "asdf-123",
      "subtitle": "Whats up",
      "title": "Test frequency",
    }
  }
  size="full">
  <Styled()
    alignItems="center"
    justifyContent="flex-start">
    <Styled()
      direction="column"
      justifyContent="center">
      <styled.h3>
        Test frequency
      </styled.h3>
      <styled.h4>
        Whats up
      </styled.h4>
    </Styled()>
  </Styled()>
  <styled.p>
    Testing the frequency profile
  </styled.p>
  <Styled()>
    <Styled()>
      Follow
    </Styled()>
  </Styled()>
  <Apollo(branch(pure(MetaDataPure)))
    id="asdf-123"
    type="frequency" />
</pure(CardPure)>
`;

exports[`test should render with a large size 1`] = `
<pure(CardPure)
  data={
    Object {
      "description": "Testing the frequency profile",
      "id": "asdf-123",
      "subtitle": "Whats up",
      "title": "Test frequency",
    }
  }
  size="large">
  <Styled()
    alignItems="center"
    justifyContent="flex-start">
    <Styled()
      direction="column"
      justifyContent="center">
      <styled.h3>
        Test frequency
      </styled.h3>
      <styled.h4>
        Whats up
      </styled.h4>
    </Styled()>
  </Styled()>
  <styled.p>
    Testing the frequency profile
  </styled.p>
  <Styled()>
    <Styled()>
      Follow
    </Styled()>
  </Styled()>
  <Apollo(branch(pure(MetaDataPure)))
    id="asdf-123"
    type="frequency" />
</pure(CardPure)>
`;

exports[`test should render with a medium size 1`] = `
<pure(CardPure)
  data={
    Object {
      "description": "Testing the frequency profile",
      "id": "asdf-123",
      "subtitle": "Whats up",
      "title": "Test frequency",
    }
  }
  size="medium">
  <Styled()
    alignItems="center"
    justifyContent="flex-start">
    <Styled()
      direction="column"
      justifyContent="center">
      <styled.h3>
        Test frequency
      </styled.h3>
      <styled.h4>
        Whats up
      </styled.h4>
    </Styled()>
  </Styled()>
  <styled.p>
    Testing the frequency profile
  </styled.p>
  <Styled()>
    <Styled()>
      Follow
    </Styled()>
  </Styled()>
</pure(CardPure)>
`;

exports[`test should render with a mini size 1`] = `
<pure(CardPure)
  data={
    Object {
      "description": "Testing the frequency profile",
      "id": "asdf-123",
      "subtitle": "Whats up",
      "title": "Test frequency",
    }
  }
  size="mini">
  <Styled()
    alignItems="center"
    justifyContent="flex-start">
    <Styled()
      direction="column"
      justifyContent="center">
      <styled.h3>
        Test frequency
      </styled.h3>
      <styled.h4>
        Whats up
      </styled.h4>
    </Styled()>
  </Styled()>
</pure(CardPure)>
`;

exports[`test should render with a small size 1`] = `
<pure(CardPure)
  data={
    Object {
      "description": "Testing the frequency profile",
      "id": "asdf-123",
      "subtitle": "Whats up",
      "title": "Test frequency",
    }
  }
  size="small">
  <Styled()
    alignItems="center"
    justifyContent="flex-start">
    <Styled()
      direction="column"
      justifyContent="center">
      <styled.h3>
        Test frequency
      </styled.h3>
      <styled.h4>
        Whats up
      </styled.h4>
    </Styled()>
  </Styled()>
  <Styled()>
    <Styled()>
      Follow
    </Styled()>
  </Styled()>
</pure(CardPure)>
`;

exports[`test should render with no size 1`] = `
<pure(CardPure)
  data={
    Object {
      "description": "Testing the frequency profile",
      "id": "asdf-123",
      "subtitle": "Whats up",
      "title": "Test frequency",
    }
  }>
  <Styled()
    alignItems="center"
    justifyContent="flex-start">
    <Styled()
      direction="column"
      justifyContent="center">
      <styled.h3>
        Test frequency
      </styled.h3>
      <styled.h4>
        Whats up
      </styled.h4>
    </Styled()>
  </Styled()>
</pure(CardPure)>
`;
